diff --git a/crates/wgsl_parser/src/syntax_kind.rs b/crates/wgsl_parser/src/syntax_kind.rs
index 281e863..8772651 100644
--- a/crates/wgsl_parser/src/syntax_kind.rs
+++ b/crates/wgsl_parser/src/syntax_kind.rs
@@ -127,7 +127,7 @@ pub enum SyntaxKind {
     UnofficialPreprocessorEndif,
     #[regex("#else.*")]
     UnofficialPreprocessorElse,
-    #[regex("#import.*")]
+    #[regex("#import")]
     UnofficialPreprocessorImport,
     #[regex("#define_import_path.*")]
     UnofficialPreprocessorDefineImportPath,
